### ISSUE TITLE
Add strict diagnostic for unknown length unpacking

### ISSUE THREAD

**Issue by dhruvmanila** [astral-sh/ty#2628]
https://github.com/astral-sh/ruff/pull/22796 changes the call checking logic to avoid raising `parameter-already-assigned` in the following case:

```py
def f(a: str, b: str, c: float): ...

# Explicit keyword argument takes precedence over variable-length variadic expansion.
# The list unpacking should only fill `a` and `b`, leaving `c` for the keyword argument.
def _(args: list[str]) -> None:
    f(*args, c=1.0)
```

But, there's an argument that we should still have a separate opt-in rule (which is also turned on in [strict mode](https://github.com/astral-sh/ty/issues/527)) to emit a diagnostic in the above case.

Relevant discussion thread: https://github.com/astral-sh/ruff/pull/22796#discussion_r2721870635

> I guess we could consider still flagging this as a separate error code? Since the length of the `args` iterable is unknown, we cannot verify that the call will _definitely_ succeed; some users who want very strict typing might still want to know about that.
>
> I agree that we should err on the side of leniency here, though -- if we did have a separate error code for that, it should definitely be disabled by default. It's too strict for most users.

> I think if we were being consistent about the strict version, it would go beyond what main currently does -- really _any_ `*` unpacking of anything other than a fixed-length tuple, to any function which doesn't take `*args`, is a call we can't verify is correct, which the strict diagnostic should be emitted on. I definitely think this is a separate (and low priority) feature, not something we should do in this PR.
